src/App.tsx

```diff
@@ -15,36 +17,112 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  ALPHABETIC = 'alphabet',
+  LENGTH = 'length',
+}
+
+type SortFunction = (a: string, b: string) => number;
+
+const sortFunctions: Record<SortType, SortFunction> = {
+  [SortType.ALPHABETIC]: (a, b) => a.localeCompare(b),
+  [SortType.LENGTH]: (good1, good2) => {
+    if (good1.length === good2.length) {
+      return good1.localeCompare(good2);
+    }
+
+    return good1.length - good2.length;
+  },
+};
+
+const getGoods = (
+  goods: string[],
+  sortType: '' | SortType,
+  isReversed: boolean,
+): string[] => {
+  const visibleGoods: string[] = [...goods];
+
+  switch (sortType) {
+    case SortType.ALPHABETIC:
+    case SortType.LENGTH:
+      visibleGoods.sort(sortFunctions[sortType]);
+      break;
+    default:
+  }
```

Comment:
```suggestion
  if (sortType && sortFunctions[sortType]) {
    visibleGoods.sort(sortFunctions[sortType]);
  }
```
- As you created a map with sort functions the switch case is redundant
- `default` case shouldn't be empty